src/complex.py

```diff
@@ -303,15 +348,258 @@ def initialize(self, points, d):
         else:
             raise Exception, 'initialize procedure only implemented for simplicial complexes with dimension 1-3.'
 
-    def insert_vertex(self, v, s):
-        pass
+    def kill_vertex(self, v=None, persist=True, kill_record=None):
+        if kill_record is not None and 'on_the_fly' in kill_record:
+            v = self.vertices[self.next_vertex_index - 1]
+            persist = True
+            kill_record = None
+        
+        if kill_record is not None:
+            self._apply_kill_record(kill_record)
+        else:
+            kill_record = {}
+            kill_record['v'] = v
+
+            if len(self.stars[v]) > 0:
+                sv = next(iter(self.stars[v]))
```

Comment:
This seems incorrect to me. The star of a vertex will be arbitrarily ordered, so your code will make an arbitrary choice over which vertex to collapse into. Correct thing is to get the full star and do an np.random.choice